api/src/main/java/io/opentelemetry/api/trace/ImmutableSpanContext.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.trace;
+
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.extension.memoized.Memoized;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+@AutoValue
+abstract class ImmutableSpanContext implements SpanContext {
```

Comment:
Since the interface is also annotated `@Immutable`  and according to the spec any SpanContext must be Immutable, this is maybe not the best name to distinguish it from possible other implementations which would also be Immutable. Maybe DefaultSpanContext?

Reply:
Yeah I had the same impression - we have `Immutable*` as a pattern in many AutoValue implementations of interfaces currently. Renaming them all to `Default` seems fine to me, or even `*Impl` so the implementation orders next to the interface in the filetree.

Reply:
No strong opinion. I'd keep this name for this PR, and we can always rename later, since this is an internal detail.